### PR TITLE
ROCm packages - Bump up the version for the rocm-5.4.3 release

### PR DESCRIPTION
**Pull request by srekolam**
This commit contains the version updates required to support the rocm-5.4.3 release.
Of the total packages (46) , this has the updates for 42 packages. only miopen-hip, miopen-opencl, migraphx, mivisionx, rocmlir are pending .

### CODE REVIEW

**Comment by eugeneswalker** [comment]
Any chance this might be ready for merge this week?

**Comment by srekolam** [comment]
added updates to remaining rocm packages - miopen, rocmlir, mivisionx,migraphx. This PR has the version updates for all rocm packages.

File: var/spack/repos/builtin/packages/miopen-opencl/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miopen-opencl/package.py:153]: Curious.  Are these changes -- removing the versions from above and adding the (same) dependencies here -- intended to make it easier to "deprecate" the older versions?  Or more for documentation purposes?

**Review by tldahlgren — DISMISSED** [review]
Confirmed all of the `version` sha256.

**Comment by tldahlgren** [comment]
Should I wait for one (or more) of the other maintainers to review or is this ready to merge (when the checks pass of course)?

File: var/spack/repos/builtin/packages/hipsparse/package.py
Comment (cgmb) [inline var/spack/repos/builtin/packages/hipsparse/package.py]: This can be further simplified. rocprim is not a direct dependency of hipsparse. It's merely a build dependency of rocsparse, so it doesn't need to be mentioned in the hipsparse package recipe at all. I would just drop the `depends_on("rocprim",  ...)` and merge the two loops into one.

**Review by cgmb — DISMISSED** [review]
My only comment was about a potential simplification. It's not a matter of correctness. If everything's building properly, LGTM.

File: var/spack/repos/builtin/packages/miopen-opencl/package.py
Comment (srekolam) [reply var/spack/repos/builtin/packages/miopen-opencl/package.py:153]: Basically, i just felt i could combine both for loops and still have the dependencies build correctly.

**Comment by eugeneswalker** [comment]
Should we rock and roll? Perhaps in the interest of having the substance here merged, we could merge and use a follow-on PR to apply simplifications suggested by @cgmb ?

File: var/spack/repos/builtin/packages/hipsparse/package.py
Comment (srekolam) [reply var/spack/repos/builtin/packages/hipsparse/package.py]: Thanks ..I have made the changes

**Review by cgmb — APPROVED** [review]

**Review by renjithravindrankannath — COMMENTED** [review]
mlirmiopen update for 5.4.3 can be included.

**Comment by srekolam** [comment]
> mlirmiopen update for 5.4.3 can be included.

rocmlir is the new recipe in place of mlirmiopen from 5.4.0 onwards

**Review by renjithravindrankannath — APPROVED** [review]
Approved.

**Comment by srekolam** [comment]
@tldahlgren , can you please let us know if this can be merged. thanks!!

**Review by alalazo — APPROVED** [review]
Two maintainers approved and pipelines passed.

**Pull request merged**
